archunit/src/test/java/com/tngtech/archunit/core/importer/ClassFileImporterSyntheticPrivateAccessesTest.java

```diff
@@ -422,6 +424,40 @@ Supplier<Supplier<Data_of_imports_private_constructor_reference_from_lambda.Targ
         );
     }
 
+    /**
+     * This is a special case, because for =+ concatenation of an outer string from an inner class
+     * the compiler creates a synthetic `access$123` method that creates a new `StringBuilder()`.
+     * Before this we wrongly assumed that all that happens from such `access$123` methods
+     * are field accesses and method calls, but no constructor calls.
```

Comment:
1. `=+` → `+=` 😉
2. Not all compilers create such a synthetic `access$123` method in this case; for this test's example, I've also seen the following byte code (e.g. with openjdk-17.0.2):
  ```
  private void stringConcat();
    Code:
       0: aload_0
       1: getfield      #1                  // Field this$0:LOuter;
       4: dup
       5: getfield      #13                 // Field Outer.outerPrivateString:Ljava/lang/String;
       8: invokedynamic #19,  0             // InvokeDynamic #0:makeConcatWithConstants:(Ljava/lang/String;)Ljava/lang/String;
      13: putfield      #13                 // Field Outer.outerPrivateString:Ljava/lang/String;
      16: return
  ```
  So I'd replace `the compiler creates` with `the compiler may create` (also in the commit message).
3. The `Before this` reference might not be clear in this JavaDoc; I would drop it here.

```suggestion
     * This is a special case, because for `+=` concatenation of an outer string from an inner class
     * the compiler may create a synthetic `access$123` method that creates a new `StringBuilder()`.
```